doc/source/configuration/ironic.rst

```diff
@@ -0,0 +1,339 @@
+          network: internal-net
+
+To provision baremetal nodes in Nova you will also require setting a flavour
+specific to that type of baremetal host. You will need to replace the custom
+resource ``resources:CUSTOM_<YOUR_BAREMETAL_RESOURCE_CLASS>`` placeholder with
+the resource class of your baremetal hosts, you will also need this later when
+configuring the baremetal-compute inventory.
+
+.. code-block:: yaml
+
+  openstack_flavors:
+    - "{{ openstack_flavor_baremetal_A }}"
+    # Bare metal compute node.
+    openstack_flavor_baremetal_A:
+    name: "baremetal-A"
+    ram: 1048576
+    disk: 480
+    vcpus: 256
+    extra_specs:
+    "resources:CUSTOM_<YOUR_BAREMETAL_RESOURCE_CLASS>": 1
+    "resources:VCPU": 0
+    "resources:MEMORY_MB": 0
+    "resources:DISK_GB": 0
+
+Enabling conntrack
+==================
+
+UEFI booting requires conntrack_helper to be configured on the Ironic neutron
+router, this is due to TFTP traffic being dropped due to being UDP. You will
+need to define some extension drivers in ``neutron.yml`` to ensure conntrack is
+enabled in neutron server.
+
+.. code-block:: yaml
+
+  kolla_neutron_ml2_extension_drivers:
+    port_security
+    conntrack_helper
+    dns_domain_ports
+
+The neutron l3 agent also requires conntrack to be set as an extension in
+``kolla/config/neutron/l3_agent.ini``
+
+.. code-block:: ini
+
+  [agent]
+  extensions = conntrack_helper
+
+It is also required to load the conntrack kernel module ``nf_nat_tftp``,
+``nf_conntrack`` and ``nf_conntrack_tftp`` on network nodes. You can load these
+modules using modprobe or define these in /etc/module-load.
+
+The Ironic neutron router will also need to be configured to use
+conntrack_helper.
+
+.. code-block:: json
+
+  "conntrack_helpers": {
+    "protocol": "udp",
+    "port": 69,
+    "helper": "tftp"
+  }
+
+Currently it's not possible to add this helper via the OpenStack CLI, to add
+this to the Ironic router you will need to make a request to the API directly,
+for example via cURL.
+
+.. code-block:: console
+
+  curl -g -i -X POST \
+  http://<internal_api_vip>:9696/v2.0/routers/<ironic_router_uuid>/conntrack_helpers \
+  -H "Accept: application/json" \
+  -H "User-Agent: openstacksdk/2.0.0 keystoneauth1/5.4.0 python-requests/2.31.0 CPython/3.9.18" \
+  -H "X-Auth-Token: <issued_token>" \
+  -d '{ "conntrack_helper": {"helper": "tftp", "protocol": "udp", "port": 69 } }'
+
+TFTP server
+===========
+
+By default the Ironic TFTP server (ironic_pxe container) will call the UEFI
+boot file ``ipxe-x86_64.efi`` instead of ``ipxe.efi`` meaning no boot file will
+be sent during the PXE boot process in the default configuration.
+
+As of now this is solved by using a hack workaround by changing the boot file
+in the ``ironic_pxe`` container. To do this you will need to enter the
+container and rename the file manually.
+
+.. code-block:: console
+
+  docker exec ironic_pxe “mv /tftpboot/ipxe-x86_64.efi /tftpboot/ipxe.efi”
```

Comment:
More a general question then a review comment, but are we looking at any ways to automate this? It sounds like we'd be relying on these to be done consistently. Some hooks in Kayobe, if not patches to Ironic upstream.

Reply:
The conntrack CLI workaround shouldn't be necessary anymore, so will need to be updated. Most of this documentation was written from working on a Train client, I don't think this workaround has been necessary since [Xena](https://github.com/openstack/python-openstackclient/commit/fa8c8d26a7696d169b0b9d5aaf6b723d8feee08a) and we can use a dedicated CLI command for this now.

I think the TFTP workaround was fixed in this [commit](https://github.com/openstack/kolla/commit/1bb57ae116b4b8931bfcd512038b436f77c67e0c) which will create a symlink for ipxe.efi to whatever architecture is used, and is in Yoga and later.

Reply:
Makes sense to have dropped these sections then, given we're targetting 2023.1 here.
Perhaps we can still save these instructions on a wiki page, just in case another pre-Yoga deployment comes along?